mq.go

```diff
@@ -84,8 +86,8 @@ func New(config Config) (MQ, error) {
 		producers:            newProducersRegistry(len(config.Producers)),
 		state:                new(int32),
 	}
+	atomic.StoreInt32(mq.state, int32(ConnectionStateUndefined))
```

Comment:
Seems like ConnectionStateDisconnected could play the role of a default state (0).